components/script/dom/raredata.rs

```diff
@@ -35,6 +36,13 @@ pub(crate) struct NodeRareData {
     pub(crate) unique_id: Option<UniqueId>,
 
     pub(crate) slottable_data: SlottableData,
+
+    /// A vector of weak references to Range instances of which the start
+    /// or end containers are this node. No range should ever be found
+    /// twice in this vector, even if both the start and end containers
+    /// are this node.
+    #[ignore_malloc_size_of = "Rc"]
+    pub(crate) ranges: Rc<WeakRangeVec>,
```

Comment:
I don't like that the range vector is reference counted now, and it doesn't make sense to me because the node is the only logical owner of this data, everyone else should get a reference to it.